Assets/Scripts/ECS/Animal/SexTypeData.cs

```diff
@@ -0,0 +1,14 @@
+﻿using System;
+using Unity.Entities;
+
+namespace Ecosystem.ECS.Animal
+{
+    /// <summary>
+    /// Specifies the sex of the animal. Male, Female
+    /// </summary>
+    [Serializable]
+    public struct SexTypeData : IComponentData
```

Comment:
I would rename this to "SexData", move the enum to be an internal class here and rename the enum to just "Sex".